cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/GoalViolationDetector.java

```diff
@@ -229,10 +235,11 @@ protected void refreshBalancednessScore(Map<Boolean, List<String>> violatedGoals
   }
 
   protected Set<String> excludedTopics(ClusterModel clusterModel) {
-    return clusterModel.topics()
-        .stream()
-        .filter(topic -> _excludedTopics.matcher(topic).matches())
-        .collect(Collectors.toSet());
+    return Utils.getTopicNamesMatchedWithPattern(_excludedTopicsPattern, clusterModel.topics());
+  }
+
+  protected Set<String> mustHaveLeaderReplicaPerBrokerTopics(ClusterModel clusterModel) {
+    return Utils.getTopicNamesMatchedWithPattern(_mustHaveLeaderReplicaPerBrokerTopicsPattern, clusterModel.topics());
```

Comment:
This is not used -- can we drop `_mustHaveLeaderReplicaPerBrokerTopicsPattern` and other changes from this class?